source/math/FloatVectorList.ooc

```diff
@@ -381,4 +381,80 @@ FloatVectorList: class extends VectorList<Float> {
 			result add(this[i] clamp(floor, ceiling))
 		result
 	}
+	findMaxIndex: func -> (Int, Float) {
+		index := 0
+		for (i in 1 .. this count)
+			if (this[index] < this[i])
+				index = i
+		(index, this[index])
+	}
+	crossCorrelation: func (other: This, vectorLength: Int) -> Float {
+		upperPart := 0.0f
+		lowerPart := 0.0f
+		lowerPart2 := 0.0f
+		for (i in 0 .. vectorLength) {
+			upperPart += ((this[i] - this mean) * (other[i] - other mean))
```

Comment:
`this mean` and `other mean` will be calculated every time you call them. Call these properties once each, before the loop, and store them in variables.